src/block/timeline/edit.js

```diff
@@ -112,6 +135,149 @@ const Edit = props => {
+
+		// gets equivalent px of 1%
+		const fillPxPercent = blockRef.current.getBoundingClientRect().height / 100
+		// converts percent to px
+		const fillPx = fillPercent * fillPxPercent
+
+		let fill = { verticalLine: fillPx, middle: fillPx }
+
+		const dot = ( middleRef.current.getBoundingClientRect().top - blockRef.current.getBoundingClientRect().top ) + topPadding
+		const branch = ( branchRef.current.getBoundingClientRect().top - blockRef.current.getBoundingClientRect().top ) + topPadding
+
+		// corrects the position of the fill for the first timeline block since the line starts at the middle
+		if ( ! previousBlock || previousBlock.getAttribute( 'data-type' ) !== 'stackable/timeline' ) {
+			fill = {
+				verticalLine: fillPx - dot,
+				middle: fillPx,
+			}
+		}
+
+		setMiddleTopPosition( { dot, branch } )
+		setFillHeight( fill )
+	}
+
+	const updateMaxHeight = ( { previousBlock, nextBlock } ) => {
+		const {
+			isFirst, isLast,
+		} = getTimelinePosition( previousBlock, nextBlock )
+
+		const {
+			hasTopPadding, topPadding,
+		} = getSelectAttributes()
+
+		let lineMaxHeight = '100%'
+		let lineTop = 0
+		const offset = topPadding + middleRef.current.getBoundingClientRect().top + ( middleRef.current.getBoundingClientRect().height / 2 ) - blockRef.current.getBoundingClientRect().top
+		if ( deviceType === 'Mobile' && isLast ) {
+			lineMaxHeight = '16px'
+		} else if ( deviceType === 'Mobile' ) {
+			lineMaxHeight = '100%'
+		} else if ( isFirst && isLast ) {
+			lineMaxHeight = '0'
+		} else if ( isFirst ) {
+			lineMaxHeight = '50%'
+			// if padding is default, use top: 50%, else use calculated value
+			lineTop = hasTopPadding ? `${ offset }px` : '50%'
+		} else if ( isLast ) {
+			lineMaxHeight = hasTopPadding ? `${ offset }px` : '50%'
+		}
+
+		setVerticalLineMaxHeight( lineMaxHeight )
+		setVerticalLineTop( lineTop )
+		handleScroll()
+	}
+
+	// update max height when device type & padding changes
+	useEffect( () => {
+		updateMaxHeight( getAdjacentBlocks() )
+	}, [ deviceType, props.attributes.blockPadding ] )
+
+	// update accent fill when anchor position or padding changes
+	useEffect( () => {
+		handleScroll()
+		document.querySelector( '.interface-interface-skeleton__content' )?.addEventListener( 'scroll', handleScroll )
+		return () => {
+			document.querySelector( '.interface-interface-skeleton__content' )?.removeEventListener( 'scroll', handleScroll )
+		}
+	}, [ props.attributes.timelineAnchor, props.attributes.blockPadding ] )
+
+	// update blocks if position changes
+	useEffect( () => {
+		const timeout = setInterval( () => {
+			const { previousBlock, nextBlock } = getAdjacentBlocks()
+
+			// check if adjacent block changes
+			if ( adjacentBlock.current?.getAttribute( 'data-type' ) !== nextBlock?.getAttribute( 'data-type' ) || adjacentBlock.current?.getAttribute( 'data-block' ) !== nextBlock?.getAttribute( 'data-block' ) ) {
```

Comment:
Please change this because of the new `nextBlock` and `previousBlock` implementation